### PR TITLE
Task/cdd 2235 list of links component

### PR DESCRIPTION
**Pull request by phill-stanley**
# Description

This PR includes an internal page links component for the composite page:

Fixes #CDD-2235

---

## Type of change

Please select the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tech debt item (this is focused solely on addressing any relevant technical debt)

---

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests at the right levels to prove my change is effective
- [x] I have added screenshots or screen grabs where appropriate
- [ ] I have added docstrings in the correct style [(google)](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings)

### CODE REVIEW

**Comment by phill-stanley** [comment]
Screenshot 2024-09-30 at 09 32 32

File: cms/dynamic_content/blocks.py [hunk 1]
```diff
@@ -126,3 +126,16 @@ def get_api_representation(cls, value, context=None) -> str | None:
             return value.full_url
 
         return None
+
+
+class PageLink(blocks.StructBlock):
+    title = blocks.CharBlock()
```
Comment (A-Ashiq) [inline cms/dynamic_content/blocks.py:132]: Is it worth adding help texts for this and the `sub_title` fields?

File: cms/dynamic_content/blocks.py [hunk 2]
```diff
@@ -126,3 +126,16 @@ def get_api_representation(cls, value, context=None) -> str | None:
...
+    sub_title = blocks.CharBlock()
```
Comment (A-Ashiq) [inline cms/dynamic_content/blocks.py:133]: Do we want the title and sub-title fields to be optional?

Comment (phill-stanley) [reply cms/dynamic_content/blocks.py]: I think the title should be required, but I can see an argument to have the sub-title filed optional for possible uses outside of the current component design.

I've updated this and caught up with Rhys to make sure front-end validation is aligned.

**Review by A-Ashiq — APPROVED** [review]

**Pull request merged**